### PR TITLE
chore: parallelize test-binary-against-recipes CI step

### LINKED ISSUE #27521: Decrease runtime for testing binary against recipes
According to [CircleCI insights](https://app.circleci.com/insights/github/cypress-io/cypress/workflows/linux-x64/jobs?branch=develop), our longest-running job is `test-binary-against-recipes` which runs tests from [cypress-example-recipes](https://github.com/cypress-io/cypress-example-recipes/tree/master) with Electron.

Here are my ideas for decreasing the duration of this job:
- Modify the [test script](https://github.com/cypress-io/cypress-example-recipes/blob/master/test-examples.js) to accept a list of example directories to run tests on 
- Parallelize the CircleCI job across 3-4 machines, and run a different set of tests on each machine

In order to do this, we need to also break out some steps into their own job, uploading the result to the CircleCI workflow in order for the assets to be used in the following jobs
- `test-binary-against-recipes`
- `test-binary-against-recipes-chrome`
- `test-binary-against-recipes-firefox`

Currently, for each of these jobs we do the following:
- Clone cypress-example-recipe repo
- Install built binary into the project

We can break this out into a common job that runs before each of these. This will allow us to parallelize the jobs as well as save time by only cloning the repo and installing the binary **one** time rather than three times per workflow run.

`test-binary-against-recipes` is the longest job in our parallel job runs, so decreasing its runtime will have an immediate and drastic effect on the overall runtime of our `linux-x64` workflow. By parallelizing these test runs, we should be able to at least cut the duration of this job in half.

### PR DESCRIPTION
**Pull request by astone123**
- Closes https://github.com/cypress-io/cypress/issues/27521

### Additional details

According to CircleCI insights, `test-binary-against-recipes` is the longest-running job that we have in our CI pipeline (it only runs on `develop` and other branches specified in `mainBuildFilters`). It's p95 runtime for the past 30 days is 25m 14s. Because it runs in parallel with the rest of the binary tasks, we need to wait for it to complete to release Cypress, re-run the workflow from failed, etc. This adds a lot of time to our development cycle.

This PR uses the existing [chunking logic](https://github.com/cypress-io/cypress-example-recipes/blob/master/test-examples.js#L24) in the `test-examples.js` script in cypress-example-recipes to split these tests across 4 machines. This brings the runtime in line with the rest of the binary steps, so we're not over-optimizing, just optimizing enough so that the runtime for all of the binary steps are within 1 minute or so of eachother.

This should significantly decrease the average runtime for our `linux-x64` workflow on `develop`.

### Steps to test

Look at the CI runs for this branch and specifically the `test-binary-against-recipes` job.

### How has the user experience changed?

n/a

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: scripts/package.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "internal-scripts",
+  "version": "0.0.0-development",
```
Comment (astone123) [inline scripts/package.json:3]: This is an un-related drive-by change. There's a warning in the console any time we run any task saying that `scripts` is missing a version. This is just to make that warning go away

Comment (jordanpowell88) [reply scripts/package.json:3]: Been wanting to do this as well

**Review by jordanpowell88 — APPROVED** [review]

**Review by chrisbreiding — APPROVED** [review]
